pkg/plugins/generic/generic_plugin_test.go

```diff
@@ -78,14 +181,165 @@ var _ = Describe("Generic plugin", func() {
 				},
 			}
 
+			hostHelper.EXPECT().WriteSwitchdevConfFile(networkNodeState, map[string]bool{"0000:00:00.0": false}).Return(false, nil)
+			needDrain, needReboot, err := genericPlugin.OnNodeStateChange(networkNodeState)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(needReboot).To(BeFalse())
+			Expect(needDrain).To(BeTrue())
+		})
+
+		It("should drain because PF link is down", func() {
+			networkNodeState := &sriovnetworkv1.SriovNetworkNodeState{
+				Spec: sriovnetworkv1.SriovNetworkNodeStateSpec{
+					Interfaces: sriovnetworkv1.Interfaces{{
+						PciAddress: "0000:00:00.0",
+						NumVfs:     1,
+						Mtu:        1500,
+						VfGroups: []sriovnetworkv1.VfGroup{{
+							DeviceType:   "netdevice",
+							PolicyName:   "policy-1",
+							ResourceName: "resource-1",
+							VfRange:      "0-1",
+							Mtu:          1500,
+						}}}},
+				},
+				Status: sriovnetworkv1.SriovNetworkNodeStateStatus{
+					Interfaces: sriovnetworkv1.InterfaceExts{{
+						PciAddress:   "0000:00:00.0",
+						NumVfs:       1,
+						TotalVfs:     1,
+						DeviceID:     "1015",
+						Vendor:       "15b3",
+						Name:         "sriovif1",
+						Mtu:          1500,
+						Mac:          "0c:42:a1:55:ee:46",
+						Driver:       "mlx5_core",
+						EswitchMode:  "legacy",
+						LinkSpeed:    "25000 Mb/s",
+						LinkType:     "IB",
+						LinkAdmState: "down",
+						VFs: []sriovnetworkv1.VirtualFunction{{
+							PciAddress: "0000:00:00.1",
+							DeviceID:   "1016",
+							Vendor:     "15b3",
+							VfID:       0,
+							Name:       "sriovif1v0",
+							Mtu:        1500,
+							Driver:     "mlx5_core",
+						}},
+					}},
+				},
+			}
+
+			hostHelper.EXPECT().WriteSwitchdevConfFile(networkNodeState, map[string]bool{"0000:00:00.0": false}).Return(false, nil)
+			needDrain, needReboot, err := genericPlugin.OnNodeStateChange(networkNodeState)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(needReboot).To(BeFalse())
+			Expect(needDrain).To(BeTrue())
+		})
+
+		It("should not drain if PF link is not down", func() {
```

Comment:
can you check what it will do for the PF link is empty? and the same for the GUID as in the code we don't return an error if we are not able to read the information

Reply:
> can you check what it will do for the PF link is empty?

That's what is tested in that check. We only take action if the link is down.

> and the same for the GUID as in the code we don't return an error if we are not able to read the information

I believe I have covered the cases below.🤔 Could you give me an example of a struct you would like me to test?
e.g. 
```
			networkNodeState := &sriovnetworkv1.SriovNetworkNodeState{
				Spec: sriovnetworkv1.SriovNetworkNodeStateSpec{
					Interfaces: sriovnetworkv1.Interfaces{{
						PciAddress: "0000:00:00.0",
						NumVfs:     1,
						Mtu:        1500,
						VfGroups: []sriovnetworkv1.VfGroup{{
							DeviceType:   "netdevice",
							PolicyName:   "policy-1",
							ResourceName: "resource-1",
							VfRange:      "0-1",
							Mtu:          1500,
							IsRdma:       true,
						}}}},
				},
				Status: sriovnetworkv1.SriovNetworkNodeStateStatus{
					Interfaces: sriovnetworkv1.InterfaceExts{{
						PciAddress:   "0000:00:00.0",
						NumVfs:       1,
						TotalVfs:     1,
						DeviceID:     "1015",
						Vendor:       "15b3",
						Name:         "sriovif1",
						Mtu:          1500,
						Mac:          "0c:42:a1:55:ee:46",
						Driver:       "mlx5_core",
						EswitchMode:  "legacy",
						LinkSpeed:    "25000 Mb/s",
						LinkType:     "ETH",
						LinkAdmState: "up",
						VFs: []sriovnetworkv1.VirtualFunction{{
							PciAddress: "0000:00:00.1",
							DeviceID:   "1016",
							Vendor:     "15b3",
							VfID:       0,
							Name:       "sriovif1v0",
							Mtu:        1500,
							Driver:     "mlx5_core",
							Mac:        "0c:42:a1:55:ee:46",
						}},
					}},
				},
			}
```